### PR TITLE
Fix issue: Tree View will show duplicated project list for multiple root scenario

### PR DESCRIPTION
**Pull request by testforstephen**
Signed-off-by: Jinbo Wang <jinbwan@microsoft.com>

Issue https://github.com/Microsoft/vscode-java-dependency/issues/125

### CODE REVIEW

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java [hunk 1]
```diff
@@ -48,4 +50,11 @@
     private static IWorkspaceRoot getWorkspaceRoot() {
         return ResourcesPlugin.getWorkspace().getRoot();
     }
+
+    // TODO Use ProjectUtils.getWorkspaceInvisibleProjectName directly when the language server is released.
+    private static String getWorkspaceInvisibleProjectName(IPath workspacePath) {
+        String fileName = workspacePath.toFile().getName();
+        String projectName = fileName + "_" + Integer.toHexString(workspacePath.toPortableString().hashCode());
```
Comment (andxu) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: return fileName + "_" + Integer.toHexString(workspacePath.toPortableString().hashCode())

**Comment by andxu** [comment]
https://github.com/Microsoft/vscode-java-dependency/pull/127

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java [hunk 2]
```diff
@@ -26,16 +27,17 @@
 import com.microsoft.jdtls.ext.core.model.PackageNode;
 
 public final class ProjectCommand {
-    public static final String DEFAULT_PROJECT_NAME = "jdt.ls-java-project";
 
     public static List<PackageNode> execute(List<Object> arguments, IProgressMonitor monitor) {
         String workspaceUri = (String) arguments.get(0);
+        IPath workspacePath = ResourceUtils.canonicalFilePathFromURI(workspaceUri);
+        String workspaceProject = getWorkspaceInvisibleProjectName(workspacePath);
```
Comment (andxu) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: workspaceProject  => invisibleProjectName

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java (hunk 1 above)
Comment (andxu) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: string.format

Comment (testforstephen) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: i think current style is more readable than string.format.

File: jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java [hunk 3]
```diff
@@ -26,16 +27,17 @@
...
+        IPath workspacePath = ResourceUtils.canonicalFilePathFromURI(workspaceUri);
+        String invisibleProjectName = getWorkspaceInvisibleProjectName(workspacePath);
 
         IProject[] projects = getWorkspaceRoot().getProjects();
         ArrayList<PackageNode> children = new ArrayList<>();
-        List<IPath> paths = Arrays.asList(ResourceUtils.filePathFromURI(workspaceUri));
+        List<IPath> paths = Arrays.asList(workspacePath);
         for (IProject project : projects) {
-            if (project.exists() && !DEFAULT_PROJECT_NAME.equals(project.getName()) ) {
+            if (project.exists() && (ResourceUtils.isContainedIn(project.getLocation(), paths) || Objects.equals(project.getName(), invisibleProjectName))) {
```
Comment (yaohaizh) [inline jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: Objects.equals is reference equals?

Comment (testforstephen) [reply jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/ProjectCommand.java]: it's an utility method which guard the null case gracefully.

**Pull request merged**